### ISSUE TITLE
Expand `programs.ssh` with options to set `IdentitiesOnly` and `StrictHostKeyChecking`

### ISSUE THREAD

**Issue by x80486** [nix-community/home-manager#4134]
## Description

I'm trying to generate the configuration I used to have in `Arch Linux` while using `programs.ssh`. The relevant configuration (`/home/x80486/.config/home-manager/home.nix`) is:

```haskell
{ config, pkgs, ... }:

{
  ...
  programs.ssh = {
    compression = true;
    enable = true;
    matchBlocks = {
      "gitlab.com" = {
        hostname = "gitlab.com";
        identityFile = "~/.ssh/gitlab-personal";
      };
      "github.com" = {
        hostname = "github.com";
        identityFile = "~/.ssh/github-personal";
      };
    };
  };
  ...
}
```

I'm unable to find a setting for `IdentitiesOnly` and `StrictHostKeyChecking`. I usually use those "across the board", so they would be placed under `Host *`...but so far the only way I've found is to use `extraConfig`.

**Comment by ncfavier** [comment]
Ditto https://github.com/nix-community/home-manager/issues/4132#issuecomment-1600914957

**Comment by rycee** [comment]
@x80486 Yeah, for now you can use extraConfig. That's the exact purpose of it 🙂

**Comment by seh** [comment]
While we're here, I also had to use the "extraConfig" attribute for setting `UserKnownHostsFile` and `ControlMaster` in "matchBlocks" entries.
